volume-providers/vpc/provider/wait_for_attach_volume.go

```diff
@@ -28,17 +28,30 @@ func (vpcs *VPCSession) WaitForAttachVolume(volumeAttachmentTemplate provider.Vo
 		return nil, err
 	}
 
-	err = vpcs.APIRetry.FlexyRetry(vpcs.Logger, func() (interface{}, error) {
-		currentVolAttachment, errAPI := vpcs.GetVolumeAttachment(volumeAttachmentTemplate)
-		return currentVolAttachment, errAPI
-	}, func(intf interface{}, err *models.Error) bool {
-		// Skip API retry logic, if there is any error keep retry as per configuration
+	var currentVolAttachment *provider.VolumeAttachmentResponse
+	err = vpcs.APIRetry.FlexyRetry(vpcs.Logger, func() (error, bool) {
+		currentVolAttachment, err = vpcs.GetVolumeAttachment(volumeAttachmentTemplate)
 		if err != nil {
-			return skipRetry(err)
+			// Need to stop retry as there is an error while getting attachment
+			// considering that vpcs.GetVolumeAttachment already re-tried
+			return err, true
 		}
-		// return true in case of volume in attached status else false for retry
-		return intf.(*provider.VolumeAttachmentResponse).Status == StatusAttached
+		// Stop retry in case of volume is attached
+		return err, currentVolAttachment != nil && currentVolAttachment.Status == StatusAttached
 	})
+	// Success case
+	if err == nil && (currentVolAttachment != nil && currentVolAttachment.Status == StatusAttached) {
```

Comment:
Here , only `err == nil` is enough. `currentVolAttachment != nil && currentVolAttachment.Status == StatusAttached)` This has already been checked inside retry/skip logic

Reply:
done!